google-cloud-tools-plugin/src/com/google/gct/idea/debugger/CloudDebuggerClient.java

```diff
@@ -101,8 +102,13 @@ private static Debugger getClient(final @Nullable String userEmail, final int ti
           user.getGoogleLoginState().addLoginListener(new LoginListener() {
             @Override
             public void statusChanged(boolean login) {
-              //aggressively remove the cached item on any status change.
-              myDebuggerClientsFromUserEmail.remove(hashkey);
+              if (!login) {
+                //aggressively remove the cached item on any status change.
+                myDebuggerClientsFromUserEmail.remove(hashkey);
+                detachCloudDebugProcesses();
+              } else {
+                // user logged in, should we do something?
```

Comment:
Whenever a new connection is requested, a new login listener is added to the user's login state which will remove the connection from the `myDebuggerClientsFromUserEmail` map. In this case `detachCloudDebugProcesses()` will be called multiple times, however once the first one finishes, any subsequent call will not find any active debug sessions, and will not do anything. Nevertheless, it looks like a bad design, so I move the detach action to the process handler class which will also register as a LoginListener, and this way all process will be detached by itself upon notification.